File: c_practice/mogura.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main(void)
{
  int num, ans, score;
  srand((unsigned int)time(NULL));
  for (int i = 0; i < 10; i++)
  {
    num = rand() % 10;
    for (int j = 0; j < 10; j++)
    {
      if (num == j)
      {
        printf("  #  ");
      }
      else
      {
        printf("     ");
      }
    }
    printf("\n");
    for (int _ = 0; _ < 10; _++)
    {
      printf(" --- ");
    }
    printf("\n");
    for (int j = 0; j < 10; j++)
    {
      printf("  %d  ", j);
    }
    printf("\n");
    scanf("%d", &ans);
    if (ans == num)
    {
      score += 1;
    };
  };
  printf("%d\n", score);
  return 0;
}

File: go_practice/mogura.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	var num, ans, score int
	rand.Seed(time.Now().UnixNano())
	for i := 0; i < 10; i++ {
		num = rand.Intn(10)
		for j := 0; j < 10; j++ {
			if j == num {
				fmt.Printf("  #  ")
			} else {
				fmt.Printf("     ")
			}
		}
		fmt.Println("")
		for j := 0; j < 10; j++ {
			fmt.Printf(" --- ")
		}
		fmt.Println("")
		for j := 0; j < 10; j++ {
			fmt.Printf("  %d  ", j)
		}
		fmt.Println("")
		fmt.Scan(&ans)
		if ans == num {
			score += 10
		}
	}
	fmt.Printf("あなたの得点は%dです。\n", score)
}
